azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -209,7 +209,7 @@ def _update(self, force_update):
         else:  # vm_settings_updated
             most_recent = vm_settings
 
-        if self._extensions_goal_state is None or most_recent.created_on_timestamp > self._extensions_goal_state.created_on_timestamp:
+        if self._extensions_goal_state is None or most_recent.created_on_timestamp >= self._extensions_goal_state.created_on_timestamp:
```

Comment:
The vmSettings ETag captures changes changes in the FT or Fabric goal states, we one can a new ETag even when the artifacts profile does not change (e.g. the svdSeqNumber -- the incarnation -- can change if there is a new Fabric goal state with new certificates). The timestamp is in the artifacts profile blob so it would not change in that case.

Adding "=" to the condition to capture the new eTag.